### PR TITLE
doc: update Zigbee weather station description

### PR DESCRIPTION
**Pull request by wiba-nordic**
The application descriptions in the nRF Connect SDK are being unified. This updates the Zigbee weather station's to match the new template. Also does some general language updates.

### CODE REVIEW

**Comment by wiba-nordic** [comment]
I made a pretty big change in the Testing section by splitting off parts into their own subheadings. I would very much like feedback on what people think about that part.

**Review by divipillai — COMMENTED** [review]
Just a minor suggestion. Rest all looks good to me.

**Comment by wiba-nordic** [comment]
Resolved merge conflict.

File: applications/zigbee_weather_station/README.rst [hunk 1]
```diff
@@ -213,25 +222,26 @@ The ``build_thingy53_nrf5340_cpuapp`` parameter specifies the output directory f
...
-1. If you want to capture packets with Wireshark, install `nRF Sniffer for 802.15.4`_ and `configure Wireshark for use with Zigbee <Configuring Wireshark for Zigbee>`_.
-   This is an optional step.
-   However, Wireshark response frame examples are provided below for reference.
-#. |connect_generic|
+After programming the application to your device, complete the following steps to test it:
+
+1. **Optional:** If you want to capture packets with Wireshark, install `nRF Sniffer for 802.15.4`_ and `configure Wireshark for use with Zigbee <Configuring Wireshark for Zigbee_>`_.
+
+   Wireshark response frame examples are provided below for reference.
+#. ``debug`` build type: |connect_generic|
```
[the remark this answers is not in the document]
Comment (wiba-nordic) [reply applications/zigbee_weather_station/README.rst:241]: `debug` build type steps are called out as being optional in the note preceding the steps. I was 50-50 on whether I should also have them specifically marked optional for each step.

**Comment by wiba-nordic** [comment]
While applying this feedback, I noticed that not all instances of mentioning the Zigbee shell sample have the sample linked. Should I add the link to all of them as well?

File: applications/zigbee_weather_station/README.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (richabp) [reply applications/zigbee_weather_station/README.rst:242]: Agree, but step 1 mentions specifically that it is optional whereas step 2 doesn't which makes it sound like mandatory.

[the remark this answers is not in the document]
Comment (wiba-nordic) [reply applications/zigbee_weather_station/README.rst:242]: Should the Wireshark step be changed into a paragraph before the steps, so it's not the first step? That way it does not imply anything for the `debug` steps.

[the remark this answers is not in the document]
Comment (richabp) [reply applications/zigbee_weather_station/README.rst:242]: Agree, can be moved to the starting note.

File: applications/zigbee_weather_station/README.rst [hunk 2]
```diff
@@ -213,25 +223,26 @@ The ``build_thingy53_nrf5340_cpuapp`` parameter specifies the output directory f
...
-After programming the application, perform the following steps to test the Zigbee weather station application on the Thingy:53 device:
+     These steps related to the ``debug`` build type mention the ``debug`` build type and are not required if you are using the ``release`` build type.
+   * The provided measurement data values depend on the Thingy:53 device's surrounding conditions.
+   * If you want to capture packets with Wireshark, install `nRF Sniffer for 802.15.4`_ and `configure Wireshark for use with Zigbee <Configuring Wireshark for Zigbee_>`_.
 
-1. If you want to capture packets with Wireshark, install `nRF Sniffer for 802.15.4`_ and `configure Wireshark for use with Zigbee <Configuring Wireshark for Zigbee>`_.
-   This is an optional step.
-   However, Wireshark response frame examples are provided below for reference.
-#. |connect_generic|
+      Wireshark response frame examples are provided below for reference.
```
Comment (richabp) [inline applications/zigbee_weather_station/README.rst:238]: Line 238: Formatting looks a bit off here.

**Review by richabp — COMMENTED** [review]
LGTM. Just one minor comment.

File: applications/zigbee_weather_station/README.rst (hunk 2 above)
Comment (wiba-nordic) [reply applications/zigbee_weather_station/README.rst]: Good catch. There was an extra space.

**Review by richabp — APPROVED** [review]

**Review by divipillai — APPROVED** [review]

**Pull request merged**